Knoema.Client/Data/TimeSeriesRawData.cs

```diff
@@ -19,10 +19,12 @@ internal void OnDeserialized(StreamingContext context)
 		{
 			Dimensions = _dimensions.Select(pair =>
 			{
+				if (pair.Value.Type != JTokenType.Object || pair.Value["key"] == null || pair.Value["name"] == null)
+					return null;
 				var value = pair.Value.ToObject<DimensionItem>();
 				value.DimensionId = pair.Key;
 				return value;
-			});
+			}).Where(d => d != null);
```

Comment:
I'd use traditional foreach loop instead of Linq to avoid Where(d => d != null). Also it seems Dimensions property is deferred set under the hood. Propose to assign array or list instead.

Reply:
done